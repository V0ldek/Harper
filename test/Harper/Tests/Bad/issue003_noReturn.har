f :: Integer -> Integer;
f n = {
  if n <= 17 {
    return 42;
  }
};

main :: Integer;
main = (f 17) + (f 18);