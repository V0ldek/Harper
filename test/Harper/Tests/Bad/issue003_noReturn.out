Error: control may reach the end of a non Unit valued function without a return statement.
During evaluation of: 
{ 
  if n <= 17 { 
   return 42 ; 
   } 
  }
Located at line 2 column 7 

Execution terminated with an error: type error.